lib/next_ls.ex

```diff
@@ -149,6 +159,30 @@ defmodule NextLS do
      )}
   end
 
+  def handle_request(
+        %TextDocumentCodeAction{
+          params: %CodeActionParams{
+            context: %CodeActionContext{diagnostics: diagnostics},
+            text_document: %TextDocumentIdentifier{uri: uri}
+          }
+        },
+        lsp
+      ) do
+    code_actions =
+      for %Diagnostic{} = diagnostic <- diagnostics,
+          check =
+            NextLS.Check.new(
+              diagnostic: diagnostic,
+              uri: uri,
+              document: lsp.assigns.documents[uri]
+            ),
+          action <- NextLS.CodeActionable.fetch(check) do
+        action
+      end
```

Comment:
so for this, I think you're going to one to loop through the diagnostics, then branch on the namespace and ask the appropriate extension to return code actions for it.

you'll also need to modify each extension to add the "data" property with the namespace. the data property is used to send data down to the client, then it sends it back up.